### ISSUE TITLE
layout has broken - everything is now left-aligned

### ISSUE THREAD

**Issue by jsms90** [elifesciences/elife-xpub#500]
### Describe the bug
For some reason, everything is now left aligned

### To Reproduce
Pull the latest changes from `develop` & start the app

### Expected behavior
How the screens are supposed to look:
![image](https://user-images.githubusercontent.com/15656538/44332731-b549d600-a464-11e8-974d-1bd12c1a7ede.png)
![image](https://user-images.githubusercontent.com/15656538/44332736-baa72080-a464-11e8-8b55-94c23e37b3c9.png)
![image](https://user-images.githubusercontent.com/15656538/44332745-c266c500-a464-11e8-8430-c51c719034a3.png)

### Screenshots
Now:
![image](https://user-images.githubusercontent.com/15656538/44332494-191fcf00-a464-11e8-9e26-b1b76fd154be.png)
![image](https://user-images.githubusercontent.com/15656538/44332534-2c329f00-a464-11e8-94a6-bcf18d4b6a7a.png)

![image](https://user-images.githubusercontent.com/15656538/44332482-12915780-a464-11e8-9b45-31f057082754.png)

**Comment by jsms90** [comment]
My first thought was that this is a `grid-styled` problem, as this is what is responsible for our layout.

Looking at the recent PRs, I'm guessing the problem was introduced in this PR: #484 "update dependencies"

Checking out an old commit (from a time when we definitely didn't have this layout issue) doesn't actually bring up the old, unbroken layout...until I re-run `yarn`. So it seems like dependency updates caused this for sure.

**Comment by diversemix** [comment]
You got #500 :1st_place_medal:  ... thanks for looking into this - seems like your best placed. :+1:

**Comment by jsms90** [comment]
No problem :smile_cat:

Well.. :flushed: ... :crossed_fingers: 

So, the layout problem was introduced by [this commit - "chore: update server dependencies "](https://github.com/elifesciences/elife-xpub/commit/51fa8c6272be327e27379532056868a6b80fbb86)

Checking out the commit beforehand, running yarn and starting the app = no layout breakages.
Checking out this commit, running yarn & starting the app = layout :cry: 

---

So. Investigation.

From the commit beforehand, running `yarn` gives this:
![image](https://user-images.githubusercontent.com/15656538/44335642-04940480-a46d-11e8-9b33-7a4694add3bb.png)

After the "update server dependencies" commit, `yarn` gives this:
![image](https://user-images.githubusercontent.com/15656538/44335664-1675a780-a46d-11e8-8087-90e77726ee66.png)

So we've got these lines, which are new:
> warning " > grid-styled@4.3.3" has unmet peer dependency "emotion@>=9.0".
> warning " > grid-styled@4.3.3" has unmet peer dependency "react-emotion@>=9.0"

Which is strange, since this was only supposed to be updating server dependencies. And at first, from the PR it looks like only server-side things have changed. But if you load the whole `yarn.lock`, you'll see that [`grid-styled` was updated from `4.1.0` to `4.3.3`](https://github.com/elifesciences/elife-xpub/commit/51fa8c6272be327e27379532056868a6b80fbb86#diff-8ee2343978836a779dc9f8d6b794c3b2R4780)

You can also see from our `yarn.lock` that grid-styled's depndencies have gone from `styled-system ">=2.0.2"` to `system-components "^2.2.3"`.

`grid-styled`'s [CHANGELOG.md](https://github.com/jxnblk/grid-styled/blob/master/CHANGELOG.md#v420---2018-06-30) says
> ### v4.2.0 - 2018-06-30
>
>  - Use system-components

But the changelog doesn't mention [versions `4.3.0`, `4.3.1`, `4.3.2` or `4.3.3`](https://www.npmjs.com/package/grid-styled).

... :woman_shrugging: 

Tried changing `"grid-styled": "^4.1.0",` to `"grid-styled": "4.1.0",` in our package.json, to force it back from 4.3.3 to 4.1.0. Ran yarn. Still got our layout problems :sweat:

**Comment by jsms90** [comment]
Hmmm.....strangely....after checking out the commit right after that: ["chore: update client side dependencies "](https://github.com/elifesciences/elife-xpub/pull/484/commits/e6d0dfb6fd6428d4ab7b44cb167f04aaccc148c2) & running `yarn`, the layout issues are gone.

Checking out the commit after that ["update dev dependencies"](https://github.com/elifesciences/elife-xpub/pull/484/commits/a527d95762c9713e70a5c81bbaf025c54f69333b) & running yarn, the same thing - layout issues are still gone.

That's all 3 commits from that PR.

So now I'm doubley confused as to what's caused this.

**Comment by jsms90** [comment]
Ok so it seems like this layout issue (as well as the `Bad auth token` problem) was introduced in PR #368 - "add uploadProgress subscription resolver"

We were fine at commit `66c00a8` - the PR to use material icons. The first commit after this point, where you can run the app - i.e. `npm run server`, is at `2529d70`. At each of the other commits in between, the server throws various errors & the app can't be started.

At commit `2529d70`, we have the layout problem & the `Bad auth token` problem. These persist in every commit since that point.

**Comment by diversemix** [comment]
Thanks for the leg work on this @jsms90

**Comment by jsms90** [comment]
Ooof this has been a doozey!

Ok, so I've narrowed it down to a change in the commit called "update pubsweet package versions". You can see that `styled-system` has been updated in the `yarn.lock` file [ from version `2.3.6` to version `3.0.2`](https://github.com/elifesciences/elife-xpub/commit/2529d7044d49b91b6c09085b5e75b9b9a6cfbd3c#diff-8ee2343978836a779dc9f8d6b794c3b2R10900).

`grid-styled` depends on `system-components`, which depends on  
`styled-system` (version `^2.3.1`)

`grid-styled` also depends on `clean-tag`, which itself depends on  
`styled-system` (version `>=2.0.0 || >=3.0.0`)

For whatever reason, the addition of version `3.0.2` has broken our use of `grid-styled`. Looking at `styled-system`'s `CHANGELOG.md`, [version `3.0` seems to have made big changes](https://github.com/jxnblk/styled-system/blob/master/CHANGELOG.md#v300)

I'm unsure what the "right" way of solving this is, since it's at the level of dependencies on dependencies? Perhaps raising an issue on `grid-styled`? But either way, I know this is the problem for sure, because if I explicitly add `"styled-system": "2.3.6"` to our package.json, then our layout is back to normal :grin:

**Comment by jsms90** [comment]
FYI, here's how I realised that `styled-system` (our of all the changes to `yarn.lock` in Mihail's commit) was the culprit.

At first, when I thought Tam's ""chore: update server dependencies" commit had introduced the bug because that one also broke the layout in the same way, `styled-system` had also changed. In that commit, [`styled-system` was also updated to `3.0.2`](https://github.com/elifesciences/elife-xpub/commit/51fa8c6272be327e27379532056868a6b80fbb86#diff-8ee2343978836a779dc9f8d6b794c3b2R10618).

In his next commit, which unbroke the layout again, [`styled-system` went from `3.0.2` back down to `2.3.6`](https://github.com/elifesciences/elife-xpub/pull/484/commits/e6d0dfb6fd6428d4ab7b44cb167f04aaccc148c2#diff-8ee2343978836a779dc9f8d6b794c3b2L10618). Hence the layout working again.

**Comment by jsms90** [comment]
Raised issues on `styled-system` (which contains `clean-tag` as a subfolder) & then on `grid-styled` itself to highlight the problem that we're having.

This morning @damnhipster and I talked about alternatives to putting `styled-system` directly in our package.json. Ideally, we wouldn't indicate that our app relies directly on `styled-system`, as this could be confusing for others down the line. Without a way to add comments to our package.json, it's not ideal to have `styled-system` sitting in there purely to hard-code a particular version, because of an error in `grid-styled`/`clean-tag`'s own package.json.

So I tried forking `clean-tag` & adjusting the package.json: https://github.com/jxnblk/styled-system/pull/270/files. The idea was then to reference this fork until such time as this is fixed in `clean-tag` itself. I've pushed this up to a branch called `fix-layout-bug` - [see the diff](https://github.com/elifesciences/elife-xpub/compare/fix-layout-bug?expand=1#diff-b9cfc7f2cdf78a7f4b91a753d10865a2R75).

But this doesn't seem to work. Our layout remains broken this way. @damnhipster identified an issue in yarn: "Allow subdirectories within git repos in yarn install" https://github.com/yarnpkg/yarn/issues/4725, which may be the reason - remember that `clean-tag` is a sub-directory inside `styled-system`.

After this go around, I've been starting to wonder whether this was such a useful approach anyway. This way, we would still be putting `clean-tag` into our package.json. The only way that this was better than my original suggestion of adding a specific version of `styled-system` directly in there, is that a link to my forked version of the npm package would make it more explicit that this was a temporary solution (even without the ability to write a comment into our package.json).

SO! We're back to the drawing board.

But, there is something in yarn called ["resolutions"](https://yarnpkg.com/lang/en/docs/selective-version-resolutions/) which may be good for this situation:
> ## Why would you want to do this? 
> You may be depending on a package that is not updated frequently, which depends on another package that got an important upgrade. In this case, if the version range specified by your direct dependency does not cover the new sub-dependency version, you are stuck waiting for the author.
